### PR TITLE
Add ActiveModel::Naming to Exchange class

### PR DESCRIPTION
**Pull request by Murph33**
Replace the instance of `Spree.t(:exchange)` with `Spree::Exchange.model_name.human` as discussed in #864.  Remove method on the class that is replaced by extending the Exchange class with ActiveModel::Naming.

### CODE REVIEW

File: core/app/models/spree/exchange.rb [hunk 1]
```diff
@@ -39,10 +40,6 @@ def self.param_key
       "spree_exchange"
     end
 
-    def self.model_name
-      Spree::Exchange
-    end
-
     def model_name
```
Comment (cbrunsdon) [inline core/app/models/spree/exchange.rb]: can this be removed too?

Comment (Murph33) [inline core/app/models/spree/exchange.rb]: I didn't have any tests fail when taking it out.  https://github.com/solidusio/solidus/pull/235, which brings it in, doesn't actually tell us why it's needed either.

Comment (cbrunsdon) [inline core/app/models/spree/exchange.rb]: Kill it.

**Comment by tvdeyen** [comment]
:+1:

File: core/spec/models/spree/exchange_spec.rb [hunk 2]
```diff
@@ -72,7 +72,7 @@ module Spree
     end
 
     describe ".model_name" do # for dom_id
```
Comment (cbrunsdon) [inline core/spec/models/spree/exchange_spec.rb]: as we no longer implement model_name ourselves I think you can safely kill this spec now

**Comment by jhawthorn** [comment]
:+1:

**Pull request merged**